contracts/rewards/src/contract/execute.rs

```diff
@@ -66,10 +67,35 @@ where
 
     pub fn update_params(
         &mut self,
-        _new_params: RewardsParams,
-        _block_height: u64,
+        new_params: RewardsParams,
+        block_height: u64,
     ) -> Result<(), ContractError> {
-        todo!()
+        let cur_epoch = self.get_current_epoch(block_height)?;
+        // If the param update reduces the epoch duration such that the current epoch immediately ends,
+        // start a new epoch at this block, incrementing the current epoch number by 1.
+        // This prevents us from jumping forward an arbitrary number of epochs, and maintains consistency for past events.
+        // (i.e. we are in epoch 0, which started at block 0 and epoch duration is 1000. At epoch 500, the params
+        // are updated to shorten the epoch duration to 100 blocks. We set the epoch number to 1, to prevent skipping
+        // epochs 1-4, and so all events prior to the start of epoch 1 have an epoch number of 0)
+        let should_end =
+            cur_epoch.block_height_started + u64::from(new_params.epoch_duration) < block_height;
+        let cur_epoch = if should_end {
+            Epoch {
+                block_height_started: block_height,
+                epoch_num: cur_epoch.epoch_num + 1,
+                rewards: new_params.rewards_per_epoch,
+            }
+        } else {
+            Epoch {
+                rewards: new_params.rewards_per_epoch,
+                ..cur_epoch
+            }
+        };
```

Comment:
```suggestion
        let cur_epoch = Epoch {
            epoch_num: cur_epoch.epoch_num + should_end as u64,
            block_height_started: if should_end {
                block_height
            } else {
                cur_epoch.block_height_started
            },
            rewards: new_params.rewards_per_epoch,
        };
```
This results in fewer lines of code but I would NOT say it's cleaner. I'll leave it up to you.

Reply:
Also, does it really hurt to always reset `block_height_started` to be the current block and always increment the `epoch_num` by 1? I feel it doesn't make any difference and would result in a simpler solution, no?

Reply:
if only rust had a ternary operator.

I don't think we can always increment the epoch num by 1. There are a variety of cases where the params update doesn't end the current epoch, in which case I don't think it makes sense to increment the epoch number (for example, the epoch duration doesn't change, the epoch duration is increasing, etc). I think it would be surprising and confusing to users, validators and developers to end the epoch in these scenarios.